test/framework/ressources/prometheus-operator-role-binding.yaml

```diff
@@ -0,0 +1,11 @@
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
Currently we test prometheus-operator with having a cluster role binding against reconciling its CRDs.

This adds an e2e test which asserts whether reconciliation still works if a) prometheus-operator is started with `--namespace` and having permissions for those namespaces only.

As of today we have no such test, this introduces it.

Reply:
To be more precise: it changes the existing allow list e2e test to use _role_ bindings only. The idea is to test against regressions in the future to prevent necessitating wider permissions than necessary.

Reply:
Great! Thanks for explaining!